src/document/printer.js

```diff
@@ -505,26 +511,14 @@ function printDocToString(doc, options) {
           break;
         }
 
-        const secondContent = parts[2];
-
-        // At this point we've handled the first pair (context, separator)
-        // and will create a new *mutable* fill doc for the rest of the content.
-        // Copying all the elements to a new array would make this algorithm quadratic,
-        // which is unusable for large arrays (e.g. large texts in JSX).
-        // https://github.com/prettier/prettier/issues/3263#issuecomment-344275152
-        let remainingDoc = doc;
-        if (doc[IS_MUTABLE_FILL]) {
-          parts.splice(0, 2);
-        } else {
-          remainingDoc = {
-            ...doc,
-            parts: parts.slice(2),
-            [IS_MUTABLE_FILL]: true,
-          };
-        }
+        const secondContent = parts.at(2);
 
         /** @type {Command} */
-        const remainingCmd = { ind, mode, doc: remainingDoc };
+        const remainingCmd = {
+          ind,
+          mode,
+          doc: { ...doc, parts: parts.slice(2) },
```

Comment:
Maybe we can simply save the the offset to `doc`?

Reply:
That's right. It looks simpler [681a603](https://github.com/prettier/prettier/pull/16595/commits/681a603c4651b8c43dd72f1f504e9bce1f01b712).